homeassistant/components/rainforest_eagle/sensor.py

```diff
@@ -95,7 +98,23 @@ async def async_setup_entry(
 ) -> None:
     """Set up a config entry."""
     coordinator = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities(EagleSensor(coordinator, description) for description in SENSORS)
+    entities = [EagleSensor(coordinator, description) for description in SENSORS]
+
+    if coordinator.data.get("zigbee:Price") not in (None, "invalid"):
+        entities.append(
+            EagleSensor(
+                coordinator,
+                SensorEntityDescription(
+                    key="zigbee:Price",
+                    name="Meter Price",
+                    native_unit_of_measurement=f"{coordinator.data['zigbee:PriceCurrency']}/{ENERGY_KILO_WATT_HOUR}",
```

Comment:
According to our docs the unit should be a currency in the ISO 4217 standard. And not have / kWh.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Dropped the device class.